src/db/options.rs

```diff
@@ -312,3 +312,35 @@ pub struct ChangeStreamPreAndPostImages {
     /// If `true`, change streams will be able to include pre- and post-images.
     pub enabled: bool,
 }
+
+/// Determines the lifetime of a cursor.
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub enum TimeoutMode {
+    /// Times out after one Iteration.
+    Iteration,
+    /// Times out after the end of the Cursor Lifetime.
+    CursorLifetime,
+}
```

Comment:
I think we should omit this (and the field below) for now. Timeout mode is only relevant for CSOT, which we don't yet support in the driver. We'll add it when we do RUST-582.